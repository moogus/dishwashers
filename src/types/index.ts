export type GridItem = {
  productId: string;
  price: string;
  title: string;
  image: string;
};

export type AdditionalServices = {
  includedServices: string[];
}

export type Product = {
  title: string;
  media: Images,
  price: string;
  productInformation: string;
  displaySpecialOffer: string;
  additionalServices: AdditionalServices;
  code: string;
}

export type Price = {
  now?: string
}

export type ApiGridItem = {
  productId: string;
  price: Price;
  title: string;
  image: string;
}

export type Images = {
  altText: string;
  urls: string[]
}

export type Media = {
  images: Images;
}

export type Details = {
  productInformation: string;
}

export type ApiProduct = {
  title : string;
  media: Media;
  price?: Price;
  details?: Details;
  displaySpecialOffer : string;
  additionalServices: AdditionalServices;
  code : string;
}