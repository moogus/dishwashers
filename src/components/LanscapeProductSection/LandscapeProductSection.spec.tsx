import React from "react";
import { render } from "@testing-library/react";
import { mocked } from "ts-jest/utils";

import useGetProduct from "../../hooks/product/get-product";
import { ProductModel } from "../../types";

import { LandscapeProductSection } from "./index";

jest.mock("../../hooks/product/get-product");
jest.mock("@reach/router");

describe("<LandscapeProductSection />", () => {
  const renderContainer = () => render(<LandscapeProductSection />);

  beforeEach(() => {
    mocked(useGetProduct).mockClear();
    mocked(useGetProduct).mockReturnValue(({
      title: "foo",
      media: {
        altText: "what the dickens",
        urls: ["foo"],
      },
      price: "Â£1000",
      details: { productInformation: "<p>hiya</p>" },
      displaySpecialOffer: "baz",
      additionalServices: { includedServices: ["stuff"] },
      code: "12345",
    } as unknown) as ProductModel);
  });

  test("component renders as expected", () => {
    const { container } = renderContainer();
    expect(container).toMatchSnapshot();
  });
});
