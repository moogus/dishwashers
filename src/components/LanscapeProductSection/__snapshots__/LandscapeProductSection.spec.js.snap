// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandscapeProductSection /> component renders as expected 1`] = `
<div>
  <div
    class="d-none d-lg-block -xl-block"
  >
    <div
      class="sc-pFZIQ hXEXXV"
    >
      <div
        class="sc-jrAGrp hBqYys"
      >
        <div
          class="row"
        >
          <div
            class="col-8  pr-0"
          >
            <div
              class="row"
            >
              <div
                class="sc-gsTCUz jyviIo col"
              >
                <img
                  alt="what the dickens"
                  class="sc-bdfBwQ eLsnTq d-flex justify-content-center"
                  data-testid="selected-image"
                  src="foo"
                />
              </div>
            </div>
            <div
              class="row"
            >
              <div
                class="col"
              >
                <div
                  class="d-flex justify-content-center"
                  data-testid="image-select-buttons"
                >
                  <button
                    class="sc-dlfnbm elhZeT"
                  >
                    •
                  </button>
                </div>
              </div>
            </div>
            <div
              class="sc-kEjbxe eArQrG row"
            >
              <div
                class="col"
              >
                <h2
                  class="sc-iBPRYJ fYRKeW"
                >
                  Product Information
                </h2>
              </div>
            </div>
            <div
              class="sc-kEjbxe eArQrG row"
            >
              <div
                class="col "
              >
                <h2
                  class="sc-fubCfw cwjpul"
                >
                  Product code: 12345
                </h2>
              </div>
            </div>
            <div
              class="sc-kEjbxe eArQrG row"
            >
              <div
                class="col"
              >
                <span />
              </div>
            </div>
          </div>
          <div
            class="col-4 border-left"
          >
            <div
              class="sc-jSgupP irYxfM row"
            >
              <div
                class="col"
              >
                <p
                  class="sc-eCssSg fXEDVA"
                >
                  £1000
                </p>
                <p
                  class="sc-gKsewC eaLtyJ"
                >
                  Claim an extra 3 years guarantee via redemption
                </p>
                <p>
                  stuff
                </p>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
